app/c2/gist.py

```diff
@@ -0,0 +1,151 @@
+import json
+import uuid
+import aiohttp
+import re
+import asyncio
+
+from base64 import b64encode
+
+from app.objects.c_c2 import C2
+from app.interfaces.c2_active_interface import C2Active
+
+
+class Gist(C2, C2Active):
+
+    def __init__(self, services, module, config, name):
+        self.key = config.get('key')
+        super().__init__(services, module, config, name)
+
+    def start(self, app):
+        loop = asyncio.get_event_loop()
+        loop.run_forever(self._default_active_c2_loop())
+
+    def get_config(self):
+        """
+        Returns this C2 objects api key
+        :return: GIST api key
+        """
+        return 'githubToken', self.key
```

Comment:
why not return the full config dict() ?

Reply:
This function was changed to config, but realistically, it's meant to be the keying material that gets encoded in the agent (i.e. API key, HTTPS Cert, etc).  It was changed to config per your comment above... but I think it should be `get_key`

Reply:
The display property is notional the "config"

Reply:
just to expand on @khyberspache response, we were shooting for a generic function that all c2 channels can implement and return keying/config information to be encoded into the agent. It may be more robust for each c2 object to have a 'dynamic_compile' function that does the actual compilation. Hopefully for the first iteration, this approach is sufficient.